### ISSUE TITLE
Extract Class Refactoring introduces compilation errors

### ISSUE THREAD

**Issue by tsantalis** [tsantalis/JDeodorant#7]
https://github.com/secure-software-engineering/soot-infoflow
Extract Class Refactoring introduces compilation errors for class
soot.jimple.infoflow.solver.cfg.InfoflowCFG

**Comment by tsantalis** [comment]
The problem is caused because the referenced enumerated type StaticFieldUse is declared private inside class InfoflowCFG.
Changing the access modifier to public partially solves the problem.
We should also avoid qualifying the constants in the switch cases.

``` java
switch (oldUse) {
    case Unknown :
    case Unused :
    case ReadWrite :
        newUse = fieldUse;
        break;
    case Read :
        newUse = (fieldUse == StaticFieldUse.Read) ? oldUse : StaticFieldUse.ReadWrite;
        break;
    case Write :
        newUse = (fieldUse == StaticFieldUse.Write) ? oldUse : StaticFieldUse.ReadWrite;
        break;
    default:
        throw new RuntimeException("Invalid field use");
}
```

**Comment by tsantalis** [comment]
Fixed in commit [67bf1d6b42423f9e753efdc4eae91654b32d961e](https://github.com/tsantalis/JDeodorant/commit/67bf1d6b42423f9e753efdc4eae91654b32d961e)
